src/main/webapp/app/core/main/controllers/createSaleDialogController.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+(function () {
+    var app = angular.module("efApp");
+
+    app.controller("CreateSaleDialogController", function CreateSaleDialogController(sales, toastr, 
+        ProductService, SalesService, $uibModalInstance) {
+        var createSaleCtrl = this;
+        createSaleCtrl.products = [];
+        createSaleCtrl.saleProducts = [];
+
+        createSaleCtrl.createSale = function createSale() {
+            var sale = prepareSale();
+            if(sale) {
+                SalesService.createSale(sale).then(function success(response) {
+                    sales.push(new Sale(response));
+                    toastr.success('Venda criada com sucesso.');
+                    $uibModalInstance.close(201);
+                }, function error(response) {
+                    toastr.error(response);
+                    $uibModalInstance.close(201);
+                });
+            } else {
+                toastr.error("A venda deve conter pelo menos um produto.");
+            }
+            
+        };
+
+        createSaleCtrl.addOrRemoveProduct = function addOrRemoveProduct(product) {
+            if(createSaleCtrl.hasProduct(product)) {
+                _.remove(createSaleCtrl.saleProducts, function(currentProduct) {
+                    return product.barCode === currentProduct.barCode;
+                });
+            } else {
+                createSaleCtrl.saleProducts.push(product);
+            }
+        };
+
+        createSaleCtrl.cancel = function cancel() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        createSaleCtrl.hasProduct = function hasProduct(product) {
+            return _.includes(createSaleCtrl.saleProducts, product);
+        };
+
+        function createSaleInstance() {
```

Comment:
createSaleCtrl.createSaleInstance = function createSaleInstance() {
would not that be correct?

Reply:
nop, I want that to be an auxiliary function, making it a private function is the best practice to achieve it